### PR TITLE
[FLINK-20343] Add new Deployment overview documentation page

### PR DESCRIPTION
**Pull request by rmetzger**
For convenience, I rendered the page here:

FireShot Capture 009 - Apache Flink 1 12-SNAPSHOT Documentation_ Clusters   Deployment - localhost

### CODE REVIEW

File: docs/ops/deployment/overview.md
Comment (XComp) [inline docs/ops/deployment/overview.md:168]: Could we match the order of the job submission modes of the picture and the sections below?

File: docs/ops/deployment/overview.md
Comment (XComp) [inline docs/ops/deployment/overview.md:154]: We should name these things in the same way where possible: The overview calls this the "job submission modes".

File: docs/ops/deployment/overview.md
Comment (XComp) [inline docs/ops/deployment/overview.md:47]: That's personal taste now, but: Could we have a bit of margin around this picture here? :-)

**Review by XComp — CHANGES_REQUESTED** [review]
Thanks for the proposal. I have a few remarks which you might want to address. Good work. 👍

File: docs/ops/deployment/overview.md
Comment (rmetzger) [reply docs/ops/deployment/overview.md:83]: I don't have a strong opinion on this one, as there is probably no right or wrong here.

I wanted to mention the three modes here already, because they are essential to understand for the resource provider detail pages.

File: docs/ops/deployment/overview.md
Comment (rmetzger) [reply docs/ops/deployment/overview.md:47]: I'll look into it ;)

File: docs/ops/deployment/overview.md
Comment (rmetzger) [reply docs/ops/deployment/overview.md:50]: Yes, I'll fix that!

File: docs/ops/deployment/overview.md
Comment (rmetzger) [reply docs/ops/deployment/overview.md:91]: I agree my current order is messed up. I basically did: order by number of users + standalone mode ;).  I'll put standalone on top.

File: docs/ops/deployment/overview.md
Comment (tillrohrmann) [inline docs/ops/deployment/overview.md:62]: Maybe
```suggestion
              Compiles batch or streaming applications into a dataflow graph, which it then submits to the JobManager.
```

File: docs/ops/deployment/overview.md
Comment (tillrohrmann) [inline docs/ops/deployment/overview.md:70]: links are missing.

File: docs/ops/deployment/overview.md
Comment (tillrohrmann) [inline docs/ops/deployment/overview.md:82]: I'd suggest to put session mode last.

File: docs/ops/deployment/overview.md
Comment (tillrohrmann) [inline docs/ops/deployment/overview.md:81]:
```suggestion
                    <li><b>Application Mode</b>: runs the cluster exclusively for one job. The job's main method (or client) gets executed on the JobManager.</li>
```

File: docs/ops/deployment/overview.md
Comment (tillrohrmann) [inline docs/ops/deployment/overview.md:82]:
```suggestion
                    <li><b>Per-Job Mode</b>: runs the cluster exclusively for one job. The job's main method (or client) runs only prior to the cluster creation.</li>
```

File: docs/ops/deployment/overview.md
Comment (tillrohrmann) [reply docs/ops/deployment/overview.md:38]: Not sure about the backticks. If we decide to go this way, then it needs to be applied consistently throughout the whole document.

File: docs/ops/deployment/overview.md
Comment (tillrohrmann) [inline docs/ops/deployment/overview.md:103]: Maybe: Flink's JobManager can be run in high availability mode which allows Flink to recover from JobManager faults. In order to failover faster, multiple standby JobManagers can be started to act as backups.

File: docs/ops/deployment/overview.md
Comment (tillrohrmann) [inline docs/ops/deployment/overview.md:124]: Maybe set anchor link if possible.

File: docs/ops/deployment/overview.md
Comment (tillrohrmann) [inline docs/ops/deployment/overview.md:159]: Should we order them 

1. application mode
2. per-job
3. application mode

File: docs/ops/deployment/overview.md
Comment (tillrohrmann) [reply docs/ops/deployment/overview.md:168]: Maybe the marketing team could help us with beautifying this picture. No offense for your artistic skills Robert!

File: docs/ops/deployment/overview.md
Comment (tillrohrmann) [inline docs/ops/deployment/overview.md:176]:
```suggestion
then all jobs running on that TaskManager (or task manager) will be affected by the failure. This, apart from a negative
```

File: docs/ops/deployment/overview.md
Comment (tillrohrmann) [inline docs/ops/deployment/overview.md:175]: Same here with "Task Manager"

File: docs/ops/deployment/overview.md
Comment (tillrohrmann) [inline docs/ops/deployment/overview.md:179]: Here we write JobManager. Hence, I'd suggest to make it consistent.

File: docs/ops/deployment/overview.md
Comment (tillrohrmann) [inline docs/ops/deployment/overview.md:207]:
```suggestion
for downloading the dependencies locally. Furthermore, it allows for more even spread of the network load for
```

File: docs/ops/deployment/overview.md
Comment (tillrohrmann) [inline docs/ops/deployment/overview.md:225]: @kl0u are we failing if we are running a multi-execute job with HA enabled?

File: docs/ops/deployment/overview.md
Comment (tillrohrmann) [inline docs/ops/deployment/overview.md:315]:
```suggestion
- **`plugins/<name>/`** are loaded at runtime by Flink through separate classloaders to avoid conflicts with classes loaded and used by Flink. Only jar files which are prepared as [plugins]({% link deployment/filesystems/plugins.md %}) can be added here.
```

File: docs/ops/deployment/overview.md
Comment (tillrohrmann) [inline docs/ops/deployment/overview.md:317]: Should this be part of the overview page? Maybe it is better suited for tips & tricks.

**Review by tillrohrmann — APPROVED** [review]
Thanks for creating this PR @rmetzger. I had a couple of comments. All in all this overview page is very good improvement over what we currently have. I'd be fine with going forward and merging it once the comments are resolved.

File: docs/ops/deployment/overview.md
Comment (rmetzger) [reply docs/ops/deployment/overview.md:38]: How about writing it as JobGraph, so that the reader notices it as a Flink terminology. Adding Backticks everywhere could disrupt the reader's flow a bit.

File: docs/ops/deployment/overview.md
Comment (rmetzger) [reply docs/ops/deployment/overview.md:70]: I wanted to wait for your restructuring PR to be merged before putting proper links.

File: docs/ops/deployment/overview.md
Comment (XComp) [reply docs/ops/deployment/overview.md:317]: I would agree that the whole Deployment Best Practices should be moved somewhere else. To me it feels like being out of scope for this page.

Comment (rmetzger) [reply docs/ops/deployment/overview.md]: I removed this part.

**Comment by rmetzger** [comment]
Thank you for your feedback!
I addressed all your comments, except the concerns regarding the figures. Since the responsible person for graphics is out sick, I don't have much hope that we will have updated graphics available soon.
Would you be okay with merging the PR as is, and pushing the nicer images through a hotfix later?

**Comment by tillrohrmann** [comment]
Sure, I think updating the figures can also happen later.

**Comment by rmetzger** [comment]
Thanks. I noticed I forgot to update some more links. I'll merge the PR once this is addressed.

**Pull request merged**
